src/Interpreters/executeQuery.cpp

```diff
@@ -1033,6 +1034,11 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
 
                 }
 
+                // InterpreterSelectQueryAnalyzer does not build QueryPlan in the constructor.
+                // We need to force to build it here to check if we need to ignore quota.
+                if (auto * interpreter_with_analyzer = dynamic_cast<InterpreterSelectQueryAnalyzer *>(interpreter.get()))
+                    interpreter_with_analyzer->getQueryPlan();
```

Comment:
How is it changed comparing to current behavior, where plan is supposed to be built now?

Reply:
`InterpreterSelectQuery` builds a plan in the constructor. `InterpreterSelectQueryAnalyzer` built a plan when `execute()` is called, but now the plan is built on this line.